app/controllers/application_controller.rb

```diff
@@ -7,10 +7,18 @@ def current_user
   end
 
   def require_login
-    return if current_user == @user
-    
-    flash[:error] = t('login.required')
-    session[:user_id] = nil
-    redirect_to root_path
+    if current_user
```

Comment:
Could you please provide a screen_record or screenshot of what is shown when user is not authorized to have access? Basically how we are handling the error

Reply:
https://github.com/meb-wtag/journey_me/assets/141820874/3a8af2e7-1f91-4273-ba5c-8d7c2d7c51db

Admin has the rights to view everything

Reply:
Could you please update the user name and display user role for the user so it's easy to understand who is the user or admin?

Also please show in the video when a user tries to update another user (supposed to throw an unauthorized error)

Reply:
https://github.com/meb-wtag/journey_me/assets/141820874/ddd31f7b-d014-45f4-ad3f-918611ca85e5

No normal user can access any other page than his own ones. If he tries to access it by routes he gets redirected to the own user-panel. So it's not even possible to try to update another user. Only admins have the right to do so

Reply:
Okay I can see the CanCan::AccessDenied exception thrown. Can you try to rescue this in the application_controller file with a suitable error message?